### PR TITLE
Show upcoming events in the sidebar

### LINKED ISSUE #13: Maya upcoming events for "not just one day" [$50]
I fell that the upcoming events sidebar should be able to show upcoming events for a longer time period.(This month or one year from now, etc)

Now it only shows the selected day. Thoughts?

Launchpad Details: [#LP1006798](https://bugs.launchpad.net/bugs/1006798) Kārlis Lukstiņš - 2012-05-31 09:25:50 +0000

### PR DESCRIPTION
**Pull request by Frenzy-Alex**
Fixes #13

Add ListBox for upcoming events, Change AllDay Event will start From 00:00:00 and ends at 23:59:59. Update Util.is_multiday_event and Util.is_all_day. Add function Util.is_event_in_range.

### CODE REVIEW

**Comment by tintou** [comment]
Hi @Frenzy-Alex thanks for your pull request.
You can't set all-day events to be from 00:00 to 23:59, all standard-compliant calendars interprets all-day events to be from 00:00 to 00:00 on day+1
(see https://icalevents.com/1778-all-day-events-adding-a-day-or-not/ )
One important thing, all-day events are timezone-independant

**Comment by Frenzy-Alex** [comment]
Hi @tintou I've update all-day events to be from 00:00 to 00:00 day+1.

**Comment by danrabbit** [comment]
I think instead of a combobox it would be better to just show all of the events and filter for each header. The combobox means you kind of have to hunt for which header might have events.

Right now you can have an empty state that looks like this, which is not really ideal:

screenshot from 2017-07-11 18 34 24

So what would probably be better is to show each of these as headers if they contain events. So for example for me currently this would look like:
```
Next Week
    Grandma's Birthday
This Month
    My Birthday
Next Month
    Taking Back Sunday
    Uncle's Birthday    
```

I'm not sure "This year" is super helpful since it would just be all the birthday's ever :p

**Comment by Frenzy-Alex** [comment]
Hi @danrabbit, which categories of upcoming events you want to see in the sidebar?

**Comment by danrabbit** [comment]
I’m not sure what you mean by categories. Can you elaborate?

**Comment by Frenzy-Alex** [comment]
I mean on which sections divide all events: Tomorrow, This Week, Next Week, This Month, Next Month.. what else?

**Comment by Frenzy-Alex** [comment]
@danrabbit, I've change ComboBox on Headers, is it okay now?

File: core/Utils.vala [hunk 1]
```diff
@@ -412,8 +445,8 @@ namespace Maya.Util {
     public bool is_multiday_event (iCal.Component comp) {
         var start = ical_to_date_time (comp.get_dtstart ());
         var end = ical_to_date_time (comp.get_dtend ());
-        start= start.to_timezone (new TimeZone.utc ());
-        end = end.to_timezone (new TimeZone.utc ());
+//        start= start.to_timezone (new TimeZone.utc ());
```
Comment (dahenson) [inline core/Utils.vala]: Please remove code instead of commenting it out. Let the diff show what used to be there.

File: src/AgendaView.vala [hunk 2]
```diff
@@ -107,31 +114,108 @@ public class Maya.View.AgendaView : Gtk.Grid {
...
-        scrolled.hscrollbar_policy = Gtk.PolicyType.NEVER;
-        scrolled.add (events_list);
+        var selected_scrolled = new Gtk.ScrolledWindow (null, null);
+        selected_scrolled.expand = true;
+        selected_scrolled.hscrollbar_policy = Gtk.PolicyType.NEVER;
+        selected_scrolled.add (selected_date_events_list);
 
-        add (day_grid);
-        add (scrolled);
+        var upcoming_events_label = new Gtk.Label ("Upcomings Events");
```
Comment (dahenson) [inline src/AgendaView.vala]: Label should be `Upcoming Events`, and needs to be translatable.

**Pull request merged**